M2/Macaulay2/packages/HomologicalShiftIdeals.m2

```diff
@@ -0,0 +1,1001 @@
+    a package for computing the homological shift ideals of a monomial ideal of a polynomial ring
+  Description
+    Text
+      HomologicalShiftIdeals is a package for computing the homological shift ideals of a monomial ideal of a polynomial ring
+///);
+
+-------------------------------------------------------------------------------------------
+-- Compute the support of a monomial ideal
+-------------------------------------------------------------------------------------------
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    supportIdeal
+    (supportIdeal,Ideal)
+  Headline
+    compute the support of a monomial ideal
+  Usage
+    supportIdeal(I)
+  Inputs
+    I:Ideal
+      a monomial ideal $I$ of a polynomial ring $S = K[x_1,\ldots,x_n]$, $K$ a field
+  Outputs
+    l:List
+      the support of $I$
+  Description
+    Text
+      The support of a monomial ideal $I$ of a polynomial ring $S=K[x_1,\ldots,x_n]$ is the set $$\text{supp}(I)=\bigcup_{u\in G(I)}\text{supp}(u),$$ where $G(I)$ is the minimal monomial generating set of $I$ and $\text{supp}(u)=\{x_j:x_j\ \text{divides}\ u\}$ for a monomial $u$.
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2,x_3,x_1^2*x_4);
+      supportIdeal(I)
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2^5,x_3,x_1*x_4);
+      supportIdeal(I)
+///);
+
+supportIdeal = method(TypicalValue=>List)
+supportIdeal (Ideal) := I -> (
+        if isMonomialIdeal(I)==false then return "expected a monomial ideal"
+        else S:=ring I;
+        n:=numgens S;  
+        GI:=flatten entries mingens I;
+        suppI:=set{};
+        for w in GI do (
+             s:=set support w;
+             suppI=suppI+s;
+        );
+        suppI
+    );
+
+-------------------------------------------------------------------------------------------
+-- Whether an ideal is fully supported
+-------------------------------------------------------------------------------------------
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    isFullySupported
+    (isFullySupported,Ideal)
+  Headline
+    whether a monomial ideal is fully supported
+  Usage
+    isFullySupported(I)
+  Inputs
+    I:Ideal
+      a monomial ideal $I$ of a polynomial ring $S = K[x_1,\ldots,x_n]$, $K$ a field
+  Outputs
+    b:Boolean
+      whether $I$ is fully supported
+  Description
+    Text
+      A monomial ideal $I$ of a polynomial ring $S=K[x_1,\ldots,x_n]$ is fully supported if $\text{supp}(I)=\{x_1,\dots,x_n\}$.
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2,x_3,x_1^2*x_4);
+      isFullySupported(I)
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_3,x_2*x_4);
+      isFullySupported(I)
+///);
+
+isFullySupported = method(TypicalValue=>Boolean)
+isFullySupported (Ideal) := I -> (
+        if isMonomialIdeal(I)==false then return "expected a monomial ideal"
+        else n:=numgens ring I;
+        suppI:=supportIdeal I;
+        if #suppI==n then return true
+        else return false
```

Comment:
These two lines can just be replaced by `#suppI == n`.

Indeed, for a boolean `x`, `if x then true else false` is always equivalent to `x`.  And a function returns the value of its final statement, so `return` isn't necessary here.